### DISCUSSION TITLE
Use ProvisioningServiceClient on MacOs

### DISCUSSION THREAD

**Discussion by hamakiri** [Azure/azure-iot-sdk-csharp#2795] [Q&A]
Hi all,

I have a question. I have written a tool to register a device with a certificate and certificate chain via the ProvisioningClient using SecurityProviderX509Certificate with the cert.pem, cert.key.pem, cert.chain.pem files which are on the disk. The cert.chain.pem contains multiple certificates (root and intermediate certificates).

For the device certificate (cert.pem) I create a X509Certificate2.CreateFromEncryptedPemFile (with the key cert.key.pem)) and for the chain I use X509Certificate2Collection and ImportFromPemFile().

I then use the ProvisioningClient with HttpTransportHandler and SecurityProviderX509Certificate (with mentioned certs) and try to RegisterAsync()

When I run it on my windows machine. Everything works fine. When I run it on a Mac M1 I get the error:
{"errorCode":401002,"trackingId":"669af3ce-469a-4b1f-b6d2-e3ef4d7ae695","message":"CA certificate not found.","timestampUtc":"2022-09-30T13:07:58.1858684Z"}

Can someone hint me what the problem could be?
Thank you!

**Comment by timtay-microsoft** [comment]
We currently don't test/support running this SDK in MacOS environments, so there isn't much help we can provide here right now.
